validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorClientService.java

```diff
@@ -274,13 +270,23 @@ public static ValidatorClientService create(
               } else {
                 LOG.error("Error was encountered during validator client service start up.", error);
               }
+              checkNoKeysLoaded(validatorConfig, validatorLoader);
               return null;
             })
         .always(() -> LOG.trace("Finished starting validator client service."));
 
     return validatorClientService;
   }
 
+  private static void checkNoKeysLoaded(
+      final ValidatorConfig validatorConfig, final ValidatorLoader validatorLoader) {
+    if (validatorConfig.isExitWhenNoValidatorKeysEnabled()
+        && validatorLoader.getOwnedValidators().getActiveValidators().isEmpty()) {
```

Comment:
As reasoned in chat, I think we should use `validatorLoader.getOwnedValidators().hasNoValidators()` here, to catch 0 loaded keys independently by their status.
In short: in my opinion VC should not exit if successfully loads keys their deposits have not been processed yet.

Reply:
agree, just getOwnedValidators checked here, not active status please...